js/id/actions/merge_remote_changes.js

```diff
@@ -0,0 +1,152 @@
+        if (!target) return;
+
+        function pointEqual(a, b) {
+            var epsilon = 1e-6;
+            return (Math.abs(a[0] - b[0]) < epsilon) && (Math.abs(a[1] - b[1]) < epsilon);
+        }
+
+        if (pointEqual(target.loc, remote.loc)) {
+            return target;
+        }
+        if (option === 'force_remote') {
+            return target.update({loc: remote.loc});
+        }
+
+        conflicts.push(t('merge_remote_changes.conflict.location'));
+        return;  // fail merge
+    }
+
+    function mergeRemoteChildren(target) {
+        if (!target) return;
+
+        if (_.isEqual(target.nodes, remote.nodes)) {
+            return target;
+        }
+        if (option === 'force_remote') {
+            return target.update({nodes: remote.nodes});
+        }
+
+        var o = base.nodes || [],
+            a = local.nodes || [],
+            b = remote.nodes || [],
+            nodes = [],
+            hunks = Diff3.diff3_merge(a, o, b, true);
+
+        for (var i = 0, imax = hunks.length; i !== imax; i++) {
+            var hunk = hunks[i];
+            if (hunk.ok) {
+                nodes.push.apply(nodes, hunk.ok);
+            }
+            else {
+                // for all conflicts, we can assume c.a !== c.b
+                // because `diff3_merge` called with `true` option to exclude false conflicts..
+                var c = hunk.conflict;
+                if (_.isEqual(c.o, c.a)) {  // only changed remotely
+                    nodes.push.apply(nodes, c.b);
+                }
+                else if (_.isEqual(c.o, c.b)) {  // only changed locally
+                    nodes.push.apply(nodes, c.a);
+                }
+                else {       // changed both locally and remotely
+                    conflicts.push(t('merge_remote_changes.conflict.nodelist'));
+                    return;  // fail merge..
+                }
+            }
+        }
+
+        return target.update({nodes: nodes});
+    }
+
+    function mergeRemoteMembers(target) {
+        if (!target) return;
+
+        if (_.isEqual(target.members, remote.members)) {
+            return target;
+        }
+        if (option === 'force_remote') {
+            return target.update({members: remote.members});
+        }
+
+        conflicts.push(t('merge_remote_changes.conflict.memberlist'));
+        return;  // fail merge
+    }
+
+    function mergeRemoteTags(target) {
+        if (!target) return;
+
+        if (_.isEqual(target.tags, remote.tags)) {
+            return target;
+        }
+        if (option === 'force_remote') {
+            return target.update({tags: remote.tags});
+        }
+
+        var keys = _.reject(_.union(_.keys(base.tags), _.keys(remote.tags)), ignoreKey),
+            tags = _.clone(target.tags),
+            changed = false,
+            fail = false;
+
+        function ignoreKey(k) {
+            return k.indexOf('tiger:') === 0 || _.contains(iD.data.discarded, k);
```

Comment:
I ignored them here because they are ignored by [Entity#hasInterestingTags](https://github.com/openstreetmap/iD/blob/master/js/id/core/entity.js#L111)